jwala-services/src/main/java/com/cerner/jwala/service/app/impl/ApplicationServiceImpl.java

```diff
@@ -130,20 +142,39 @@ public Application getApplication(final String name) {
 
     @Transactional
     @Override
-    public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser) {
+    public Application updateApplication(UpdateApplicationRequest updateApplicationRequest, User anUpdatingUser)
+            throws Exception{
         updateApplicationRequest.validate();
-
         final Application application = applicationPersistenceService.updateApplication(updateApplicationRequest);
+
+        String appName = application.getName();
+        Identifier<Group> newGroupId = application.getGroup().getId();
+        Long id = newGroupId.getId();
+
+        List<Long> idList = new ArrayList<>();
+        idList.add(id);
+        List<JpaGroup> jpaGroups = groupPersistenceService.findGroups(idList);
+        if (jpaGroups.size() == 1) {
+            JpaApplication jpaApp = applicationPersistenceService.getJpaApplication(appName);
+            resourceDao.updateResourceGroup(jpaApp, jpaGroups.get(0));
+        }
+        else {
+            throw new Exception("One Jpa Group expected for the application.");
```

Comment:
Don't use the base Exception class here. Instead use the ApplicationServiceException.